### DISCUSSION TITLE
Unique `id` for Repository on `/repos/` API endpoint

### DISCUSSION THREAD

**Discussion by humitos** [community/community#29784] [Apps, API and Webhooks]
Hi! We use the `/repos/` (https://docs.github.com/en/rest/repos/repos) endpoint from the API to save a local cache to speed up some queries for our application without hitting GitHub API. 

After some time we found some weird behavior where some repositories were duplicated in our local cache. In the last months, it started to happen more frequently and while debugging it I found that the `id` returned by this endpoint was _not unique_ and could change over time --even if all the rest of its attribute are the same.

As an example, we detected that on August 16, 2022 the repository `extratone/wtf` changes its `id` from `422691395` to `515187740`:

```
$ gh api -H "Accept: application/vnd.github+json" /repos/extratone/wtf | jq | head      
{
  "id": 515187740,             // this is the `id` we are using as unique
  "node_id": "R_kgDOHrUkHA",
  "name": "wtf",
  "full_name": "extratone/wtf",
  "private": false,
  "owner": {
    "login": "extratone",
    "id": 43663476,
    "node_id": "MDQ6VXNlcjQzNjYzNDc2",
```

What is the reason behind this `id` change? **What is the field from the `/repos/` endpoint we should consider unique?**

Note that we are tracking it by `id` and not by `full_name` because the later could change if the repository is renamed or transferred to a different organization.

**Comment by adelinn** [comment]
I am also interested whether the repository id is unique and immutable or not.

**Comment by typed-sigterm** [comment]
So guys let's use `node_id`.

Bump ⬆️ also interested in this.

**Comment by marckhair** [comment]
Didn't test it for repos, but the `node_id` of branches changes on branch rename. Might be globally unique, but it does not seem immutable.

**Reply by Miggleness** [reply]
I believe it's because behind the scenes that deletes the existing branch and creates a new one.

**Comment by heronsanches** [comment]
We may want to use "full_name" as the unique id.

**Reply by typed-sigterm** [reply]
No `full_name` will change when the repo is renamed, transferred or the owner is renamed
